querybuilder/type_convert.go

```diff
@@ -13,98 +13,107 @@ const (
 )
 
 // String
-func toString(v interface{}) string {
-	switch v.(type) {
+func toString(v interface{}) (interface{}, interface{}) {
+	var err interface{} = nil
+	switch v := v.(type) {
 	case string:
-		return v.(string)
+		return v, err
 	case float64:
-		return fmt.Sprintf("%f", v.(float64))
+		return fmt.Sprintf("%f", v), err
 	case bool:
-		return fmt.Sprintf("%t", v.(bool))
+		return fmt.Sprintf("%t", v), err
 	default:
-		return ""
+		return "", err
 	}
 }
 
 // Double
-func toDouble(v interface{}) float64 {
-	switch v.(type) {
+func toDouble(v interface{}) (interface{}, interface{}) {
+	var err interface{} = nil
+	switch v := v.(type) {
 	case string:
-		f, _ := strconv.ParseFloat(v.(string), 64)
-		return f
+		f, err := strconv.ParseFloat(v, 64)
+		return f, err
 	case float64:
-		return v.(float64)
+		return v, err
 	default:
-		return 0
+		return 0, err
 	}
 }
 
 // Integer
-func toInteger(v interface{}) int {
-	switch v.(type) {
+func toInteger(v interface{}) (interface{}, interface{}) {
+	var err interface{} = nil
+	switch v := v.(type) {
 	case string:
-		i, _ := strconv.Atoi(v.(string))
-		return i
+		i, err := strconv.Atoi(v)
+		return i, err
 	case float64:
-		return int(v.(float64))
+		return int(v), err
+	case int:
+		return v, err
 	case bool:
-		if v.(bool) {
-			return 1
+		if v {
+			return 1, err
 		}
-		return 0
+		return 0, err
 	default:
-		return 0
+		return 0, err
 	}
 }
 
 // Boolean
-func toBoolean(v interface{}) bool {
-	switch v.(type) {
+func toBoolean(v interface{}) (interface{}, interface{}) {
+	var err interface{} = nil
+	switch v := v.(type) {
 	case string:
-		b, _ := strconv.ParseBool(v.(string))
-		return b
+		b, err := strconv.ParseBool(v)
+		return b, err
```

Comment:
```suggestion
		return strconv.ParseBool(v)
```

Reply:
These sort of statements can be written in 1 line. There are other places as well